sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/__init__.py

```diff
@@ -0,0 +1,195 @@
+        """
+        filtered_envelopes = []
+        for envelope in envelopes:
+            accepted = True
+            for processor in self._telemetry_processors:
+                try:
+                    if processor(envelope) is False:
+                        accepted = False
+                        break
+                except Exception as ex:
+                    logger.warning("Telemetry processor failed with: %s.", ex)
+            if accepted:
+                filtered_envelopes.append(envelope)
+        return filtered_envelopes
+
+    def _transmit_from_storage(self) -> None:
+        for blob in self.storage.gets():
+            # give a few more seconds for blob lease operation
+            # to reduce the chance of race (for perf consideration)
+            if blob.lease(self.options.timeout + 5):
+                envelopes = blob.get()
+                result = self._transmit(envelopes)
+                if result == ExportResult.FAILED_RETRYABLE:
+                    blob.lease(1)
+                else:
+                    blob.delete()
+
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-nested-blocks
+    # pylint: disable=too-many-return-statements
+    def _transmit(self, envelopes: typing.List[TelemetryItem]) -> ExportResult:
+        """
+        Transmit the data envelopes to the ingestion service.
+
+        Returns an ExportResult, this function should never
+        throw an exception.
+        """
+        if len(envelopes) > 0:
+            try:
+                track_response = self.client.track(envelopes)
+                if not track_response.errors:
+                    logger.info("Transmission succeeded: Item received: %s. Items accepted: %s",
+                                track_response.items_received, track_response.items_accepted)
+                    return ExportResult.SUCCESS
+                else:
+                    resend_envelopes = []
+                    for error in track_response.errors:
+                        if self._is_retryable_code(error.statusCode):
+                            resend_envelopes.append(
+                                envelopes[error.index]
+                            )
+                        else:
+                            logger.error(
+                                "Data drop %s: %s %s.",
+                                error.statusCode,
+                                error.message,
+                                envelopes[error.index],
+                            )
+                    if resend_envelopes:
+                        self.storage.put(resend_envelopes)
+
+            except HttpResponseError as response_error:
+                if self._is_retryable_code(response_error.status_code):
+                    return ExportResult.FAILED_RETRYABLE
+                else:
+                    return ExportResult.FAILED_NOT_RETRYABLE
+            except Exception as ex:
```

Comment:
Customers complain about seeing this error in the case of ingestion timeout (which is common). We should have the `except requests.Timeout` catch here before the broad exception catch.

```
            except requests.Timeout:
                logger.warning(
                    "Request time out. Ingestion may be backed up. Retrying."
                )
                return ExportResult.FAILED_RETRYABLE
```

Reply:
We are not using requests library anymore and we rely on Azure core pipeline for the actual HTTP calls, timeout must trigger an HttpResponseError with reponse_error as 408